WooCommerce/WooCommerceTests/Model/MarkOrderAsReadUseCaseTests.swift

```diff
@@ -0,0 +1,168 @@
+    }
+    private lazy var sampleNotes: [Yosemite.Note] = {
+        return try! mapNotes(from: "notifications-load-all")
+    }()
+
+    private func sampleNote(read: Bool) -> Yosemite.Note? {
+        return sampleNotes.first { note in
+            return note.read == read && note.meta.identifier(forKey: .order) != nil
+        }
+    }
+
+    override func setUp() {
+        super.setUp()
+        dispatcher = Dispatcher()
+        storageManager = MockStorageManager()
+        storesManager = MockStoresManager(sessionManager: .makeForTesting())
+        network = MockNetwork()
+
+        NotificationStore.resetSharedDerivedStorage()
+    }
+
+    override func tearDown() {
+        NotificationStore.resetSharedDerivedStorage()
+        super.tearDown()
+    }
+
+    private func setupStoreManagerReceivingNotificationActions(for note: Yosemite.Note, noteStore: NotificationStore) {
+        storesManager.whenReceivingAction(ofType: NotificationAction.self) { action in
+            switch action {
+            case let .synchronizeNotifications(onCompletion):
+                onCompletion(nil)
+            case let .synchronizeNotification(_, onCompletion):
+                onCompletion(note, nil)
+            case let .updateReadStatus(noteID, read, onCompletion):
+                noteStore.updateLocalNoteReadStatus(for: [noteID], read: read) {
+                    onCompletion(nil)
+                }
+            default:
+                XCTFail("Unexpected action: \(action)")
+            }
+        }
+    }
+
+    @MainActor
+    func test_markOrderNoteAsReadIfNeeded_with_stores_unreadNote() throws {
+        let unreadNote = try XCTUnwrap(sampleNote(read: false))
+        let orderID = try XCTUnwrap(unreadNote.meta.identifier(forKey: .order))
+
+        let expectation = expectation(description: "Mark order as read with stores")
+        let noteStore = NotificationStore(dispatcher: dispatcher, storageManager: storageManager, network: network)
+
+        setupStoreManagerReceivingNotificationActions(for: unreadNote, noteStore: noteStore)
+
+        noteStore.updateLocalNotes(with: [unreadNote]) {
+            XCTAssertEqual(self.viewStorage.countObjects(ofType: Storage.Note.self), 1)
+            Task {
+                let result = await MarkOrderAsReadUseCase.markOrderNoteAsReadIfNeeded(stores: self.storesManager, noteID: unreadNote.noteID, orderID: orderID)
+                switch result {
+                case .success(let markedNote):
+                    XCTAssertEqual(unreadNote.noteID, markedNote.noteID)
+                    let storageNote = self.viewStorage.loadNotification(noteID: markedNote.noteID)
+                    XCTAssertEqual(storageNote?.read, true)
+                    expectation.fulfill()
+                case .failure(let error):
+                    XCTFail(error.localizedDescription)
+                }
+            }
+        }
+
+        wait(for: [expectation], timeout: Constants.expectationTimeout)
```

Comment:
@bozidarsevo I haven't tried it on my local, but think the async await patter could be applied to this test, too?

Reply:
Problem is that `updateLocalNotes` function which uses completion block so I need to wait for it the be called to continue.

Reply:
One option here could be to use a continuation in the completion block. Something like this:

```
await withCheckedContinuation { continuation in
    noteStore.updateLocalNotes(with: [unreadNote]) {
        XCTAssertEqual(self.viewStorage.countObjects(ofType: Storage.Note.self), 1)
        continuation.resume()
    }
}

let result = await MarkOrderAsReadUseCase.markOrderNoteAsReadIfNeeded(stores: self.storesManager, noteID: unreadNote.noteID, orderID: orderID)

switch result {
case .success(let markedNote):
    XCTAssertEqual(unreadNote.noteID, markedNote.noteID)
    let storageNote = self.viewStorage.loadNotification(noteID: markedNote.noteID)
    XCTAssertEqual(storageNote?.read, true)
case .failure(let error):
    XCTFail(error.localizedDescription)
}
```

Reply:
Totally forgot about that and I used it in the tests already 🙈
Will update it! 👍